zproject/backends.py

```diff
@@ -936,15 +936,25 @@ def dict_representation(cls) -> List[ExternalAuthMethodDictT]:
             signup_url=reverse('login-sso'),
         )]
 
-def redirect_deactivated_user_to_login() -> HttpResponseRedirect:
+def redirect_to_signup(realm: Realm) -> HttpResponseRedirect:
+    login_url = reverse('register')
+    redirect_url = realm.uri + login_url
+    return HttpResponseRedirect(redirect_url)
+
+def redirect_to_login(realm: Realm) -> HttpResponseRedirect:
+    login_url = reverse('zerver.views.auth.login_page', kwargs = {'template_name': 'zerver/login.html'})
```

Comment:
At some point we should look at whether we can get rid of `/accounts/login` as a URL; it's not user-facing so it seems like we should be able to eliminate it.  But that can be a follow-up.

Reply:
To clarify, you mean getting rid of these
```
    # We have two entries for accounts/login; only the first one is
    # used for URL resolution.  The second here is to allow
    # reverse("django.contrib.auth.views.login") in templates to
    # return `/accounts/login/`.
    url(r'^accounts/login/$', zerver.views.auth.login_page,
        {'template_name': 'zerver/login.html'}, name='zerver.views.auth.login_page'),
    url(r'^accounts/login/$', LoginView.as_view(template_name='zerver/login.html'),
        name='django.contrib.auth.views.login'),
```

and using only ``/login`` for displaying the login_page?

Reply:
Yeah.  I'm not sure where we actually use /accounts/login, but it seems like it should be unnecessary.

Reply:
Yup, I was confused by why we had that endpoint duplicated with /login/. Could be good to remove this as a followup.

Reply:
I think there's probably some ancient history here where we needed `reverse('django.contrib.auth.views.login')` or similar to work, perhaps due to a third-party library.  But I doubt that's still something we need to do, so it'd be nice to clean this up.